src/components/toogle-switch/toggle-switch.tsx

```diff
@@ -71,26 +65,51 @@ const Slider = styled.span`
     -ms-transform: translateX(11px);
     transform: translateX(11px);
   }
+
+  ${Input}:disabled + &::before {
+    background-color: ${color_control_disabled};
+  }
 `;
 
-/**
- * TODO: Add types to component
- */
+const Switch = styled.div<{ disabled?: boolean }>`
+  position: relative;
+  width: 28px;
+  height: 18px;
+  ${Label} {
+    ${({ disabled }) =>
+      !disabled &&
+      css`
+        cursor: pointer;
+      `}
+  }
+`;
+
+type ToggleSwitchProps = {
+  checked: boolean;
+  onChange: () => void;
+  name?: string;
+  id?: string;
+  title?: string;
+  disabled?: boolean;
+};
+
 export const ToggleSwitch = ({
   checked,
   onChange,
   name = "",
   id = "",
   title = "",
-}) => {
+  disabled = false,
+}: ToggleSwitchProps) => {
   return (
-    <Switch>
+    <Switch disabled={disabled}>
```

Comment:
Disabled button should not react on hover

Reply:
Done.